### PR TITLE
[CodeQuality] Create InArrayAndArrayKeysToArrayKeyExistsRector

### PR DESCRIPTION
**Pull request by carusogabriel**
As suggested in https://github.com/rectorphp/rector/issues/424#issuecomment-383379319.

Reference that the two logics are the same: https://3v4l.org/2ok4M

### CODE REVIEW

**Comment by TomasVotruba** [comment]
New PR :tada:  

Shall I review? :)

**Comment by carusogabriel** [comment]
@TomasVotruba Please. I guess we can already start this new level with subfolders (sublevels), but I'd like your advice

File: src/Rector/Contrib/CodeQuality/InArrayAndArrayKeysToArrayKeyExistsRector.php [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        }
+
+        /** @var Name $funcCallName */
+        $funcCallName = $node->name;
+
+        if (! $funcCallName instanceof Name) {
+            return false;
+        }
+
+        if ($funcCallName->toString() !== 'in_array') {
```
Comment (TomasVotruba) [inline src/Rector/Contrib/CodeQuality/InArrayAndArrayKeysToArrayKeyExistsRector.php]: What about shorter?

```php
return $funcCallName->toString() === 'in_array';
```

**Comment by TomasVotruba** [comment]
> I guess we can already start this new level with subfolders (sublevels), but I'd like your advice

I'd wait until we have 20 or so rules. Now it has no added value

File: src/Rector/Contrib/CodeQuality/InArrayAndArrayKeysToArrayKeyExistsRector.php (hunk 1 above)
Comment (carusogabriel) [reply src/Rector/Contrib/CodeQuality/InArrayAndArrayKeysToArrayKeyExistsRector.php]: I'd like to keep the early returns. Is there any problem? :)

Comment (TomasVotruba) [reply src/Rector/Contrib/CodeQuality/InArrayAndArrayKeysToArrayKeyExistsRector.php]: Is this early return?

Comment (carusogabriel) [reply src/Rector/Contrib/CodeQuality/InArrayAndArrayKeysToArrayKeyExistsRector.php]: The above are

Comment (TomasVotruba) [reply src/Rector/Contrib/CodeQuality/InArrayAndArrayKeysToArrayKeyExistsRector.php]: Above calls are fine :)

Just `if ($funcCallName->toString() !== 'in_array') {` is not neccesary. 
I recall you've send similar PRs to code here to get rid of there `$value ? true : false` calls

**Comment by TomasVotruba** [comment]
Thank you, well done :+1:

**Pull request merged**